### PR TITLE
[Page Header] New page header for applications and UI updates

### PR DESCRIPTION
**Pull request by TackAdam**
### Description
1. Updates the applications UI and registers the create button to the header.
2. Removes Logs from the application options while under new navigation as it is not available. 

Old Applications under new navigation:
Screenshot 2024-08-21 at 11 11 48 AM
After changes:
Screenshot 2024-08-21 at 2 34 43 PM

Old applications without new navigation:
Screenshot 2024-08-21 at 11 13 40 AM
After changes: 
Screenshot 2024-08-22 at 2 49 06 PM

Video of new actions working (removed button and added to table):

https://github.com/user-attachments/assets/fc0a929a-3727-407a-baa6-5cbf83fdfc40

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/constants/shared.ts [hunk 1]
```diff
@@ -139,7 +139,6 @@ export const LONG_CHART_COLOR = PLOTLY_COLOR[1];
 export const pageStyles: CSS.Properties = {
   float: 'left',
   width: '100%',
```
Comment (ps48) [inline common/constants/shared.ts:141]: Are these needed anymore. Cause when we removed the mac width the other two css lines became irrelevant.

File: public/plugin_headerControl.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        )}
+      </>
+    );
+  }
+
+  // Only render the components if the nav group is disabled
+  return (
+    <>
+      {components.map((component, index) =>
+        React.cloneElement(component, { key: `component-fallback-${index}` })
```
Comment (ps48) [inline public/plugin_headerControl.tsx:68]: Why do we need clone element here? Also, why do we have this key named as `component-fallback`

File: public/components/application_analytics/components/application.tsx [hunk 3]
```diff
@@ -512,16 +519,21 @@ export function Application(props: AppDetailProps) {
...
+                </EuiTitle>
+                <EuiText>
+                  <p>{application.description}</p>
+                </EuiText>
+              </>
+            )}
+            {newNavigation && (
+              <HeaderControlledComponentsWrapper description={application.description} />
+            )}
+          </EuiPageHeaderSection>
```
Comment (ps48) [inline public/components/application_analytics/components/application.tsx:536]: Nit. Would this be cleaner? 

```
     {newNavigation ? 
        (<HeaderControlledComponentsWrapper description={application.description} />) 
       :  (
              <>
                <EuiTitle size="l">
                  <h1 data-test-subj="applicationTitle">{application.name}</h1>
                </EuiTitle>
                <EuiText>
                  <p>{application.description}</p>
                </EuiText>
              </>
            )}
            
```

File: public/components/application_analytics/components/app_table.tsx [hunk 4]
```diff
@@ -248,59 +213,69 @@ export function AppTable(props: AppTableProps) {
...
+
+  const filteredApplications = applications.filter((app) =>
+    app.name.toLowerCase().includes(searchQuery.toLowerCase())
+  );
 
   return (
-    <div style={pageStyles}>
+    <>
+      {!newNavigation && (
+        <div style={{ paddingLeft: '16px', paddingTop: '8px' }}>
```
Comment (ps48) [inline public/components/application_analytics/components/app_table.tsx:229]: Can may be remove these custom styles.

Comment (TackAdam) [reply public/components/application_analytics/components/app_table.tsx:229]: Using EUI page to add padding adds additional white space, Have not found a good alternative.
Screenshot 2024-08-22 at 1 23 36 PM

File: public/components/application_analytics/components/application.tsx (hunk 3 above)
Comment (TackAdam) [reply public/components/application_analytics/components/application.tsx]: Yes, have adjusted thank you

File: public/plugin_headerControl.tsx (hunk 2 above)
Comment (TackAdam) [reply public/plugin_headerControl.tsx]: Removed now just using
`  return <>{components}</>;`

File: common/constants/shared.ts (hunk 1 above)
Comment (TackAdam) [reply common/constants/shared.ts:141]: Removed pageStyle, was also used in notebooks so was going to do it separately but removed from both.

**Review by ps48 — APPROVED** [review]

**Review by Swiddis — APPROVED** [review]

**Pull request merged**